src/main/python/ttconv/imsc/attributes.py

```diff
@@ -297,6 +298,31 @@ class TemporalAttributeParsingContext:
   frame_rate: Fraction = Fraction(30, 1)
   tick_rate: int = 1
 
+@dataclass
+class TemporalAttributeWritingContext:
+  frame_rate: Fraction = Fraction(30, 1)
+  tick_rate: int = 1
+
+def to_time_format(context: TemporalAttributeWritingContext, time: Fraction) -> str:
+  if context.frame_rate.numerator == 1 and context.frame_rate.denominator == 1:
+    return to_hh_mm_ss_ms(time)
+  return to_frames(context, time)
+
+def to_hh_mm_ss_ms(time: Fraction) -> str:
+  value = time.numerator / time.denominator
+  value = datetime.timedelta(seconds = value)
```

Comment:
Currently returns microseconds in the last field. Do we want milliseconds?

Reply:
Yes, we should round to 3 digits, e.g. `round(x, 3)`

Reply:
Done